firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvent.kt

```diff
@@ -59,6 +59,41 @@ internal data class SessionInfo(
   /** What order this Session came in this run of the app. For the first Session this will be 0. */
   val sessionIndex: Int,
 
+  /** Tracks when the event was initiated */
+  var eventTimestampUs: Long,
+
+  /** Data collection status of the dependent product SDKs. */
+  var dataCollectionStatus: DataCollectionStatus = DataCollectionStatus(),
+
   /** Identifies a unique device+app installation: go/firebase-installations */
   var firebaseInstallationId: String = "",
 )
+
+/** Contains the data collection state for all dependent SDKs and sampling info */
+internal data class DataCollectionStatus(
+  val performance: DataCollectionState = DataCollectionState.COLLECTION_ENABLED,
+  val crashlytics: DataCollectionState = DataCollectionState.COLLECTION_ENABLED,
+  val sessionSamplingRate: Double = 1.0,
```

Comment:
Interesting to have sessionSamplingRate inside the `DataCollectionStatus`. I was thinking DataCollectionStatus is for external dependencies and used sessionSamplingRate as a remote config. But this looks a bit different from that intention. Curious to hear your rationale to have session sampling rate inside DataCollectionStatus.

Reply:
This is how it is in our proto - 
 I wanted to be consistent with how it's defined: https://source.corp.google.com/piper///depot/google3/logs/proto/firebase/appquality/sessions/session_event.proto;l=67?q=DataCollectionStatus